### PR TITLE
feat: Add `create_news_channel` method.

### PR DESCRIPTION
**Pull request by benwoo1110**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition
- [ ] Tests change

## Description

there is a create methods for all channel types, news channel was missing, so I added it.

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/models/discord/guild.py [hunk 1]
```diff
@@ -755,6 +755,45 @@ async def create_text_channel(
...
+        name: str,
+        topic: Absent[Optional[str]] = MISSING,
+        position: Absent[Optional[int]] = MISSING,
+        permission_overwrites: Absent[
+            Union[dict, "models.PermissionOverwrite", List[Union[dict, "models.PermissionOverwrite"]]]
+        ] = MISSING,
+        category: Union[Snowflake_Type, "models.GuildCategory"] = None,
+        nsfw: bool = False,
+        reason: Absent[Optional[str]] = MISSING,
+    ) -> "models.GuildText":
```
Comment (leestarb) [inline dis_snek/models/discord/guild.py]: damn

**Review by leestarb — CHANGES_REQUESTED** [review]
Wrong typehint

File: dis_snek/models/discord/guild.py (hunk 1 above)
Comment (benwoo1110) [reply dis_snek/models/discord/guild.py]: loll, sometimes i wish python is statically typed

**Review by leestarb — APPROVED** [review]
(Untested)

File: dis_snek/models/discord/channel.py [hunk 2]
```diff
@@ -1231,6 +1231,42 @@ async def create_text_channel(
...
+        topic: Absent[Optional[str]] = MISSING,
+        position: Absent[Optional[int]] = MISSING,
+        permission_overwrites: Absent[
+            Union[dict, PermissionOverwrite, List[Union[dict, PermissionOverwrite]]]
+        ] = MISSING,
+        nsfw: bool = False,
+        reason: Absent[Optional[str]] = MISSING,
+    ) -> "GuildText":
+        """
+        Create a text channel in this guild within this category.
```
Comment (silasary) [inline dis_snek/models/discord/channel.py]: Probably want to call it a news channel in the docstring

File: dis_snek/models/discord/guild.py [hunk 3]
```diff
@@ -755,6 +755,45 @@ async def create_text_channel(
...
+        position: Absent[Optional[int]] = MISSING,
+        permission_overwrites: Absent[
+            Union[dict, "models.PermissionOverwrite", List[Union[dict, "models.PermissionOverwrite"]]]
+        ] = MISSING,
+        category: Union[Snowflake_Type, "models.GuildCategory"] = None,
+        nsfw: bool = False,
+        reason: Absent[Optional[str]] = MISSING,
+    ) -> "models.GuildNews":
+        """
+        Create a text channel in this guild.
```
Comment (silasary) [inline dis_snek/models/discord/guild.py]: As above

**Review by silasary — APPROVED** [review]

File: dis_snek/models/discord/guild.py (hunk 3 above)
Comment (benwoo1110) [reply dis_snek/models/discord/guild.py]: AAAAAAAAAAAAAAAAA

File: dis_snek/models/discord/channel.py [hunk 4]
```diff
@@ -1203,7 +1203,7 @@ async def create_text_channel(
         nsfw: bool = False,
         rate_limit_per_user: int = 0,
         reason: Absent[Optional[str]] = MISSING,
-    ) -> "GuildText":
```
Comment (silasary) [inline dis_snek/models/discord/channel.py]: This is a GuildText

File: dis_snek/models/discord/channel.py [hunk 5]
```diff
@@ -1231,6 +1231,42 @@ async def create_text_channel(
...
+        self,
+        name: str,
+        topic: Absent[Optional[str]] = MISSING,
+        position: Absent[Optional[int]] = MISSING,
+        permission_overwrites: Absent[
+            Union[dict, PermissionOverwrite, List[Union[dict, PermissionOverwrite]]]
+        ] = MISSING,
+        nsfw: bool = False,
+        reason: Absent[Optional[str]] = MISSING,
+    ) -> "GuildText":
```
Comment (silasary) [inline dis_snek/models/discord/channel.py]: The GuildNews is down here

**Review by silasary — CHANGES_REQUESTED** [review]

**Review by silasary — APPROVED** [review]

**Pull request merged**